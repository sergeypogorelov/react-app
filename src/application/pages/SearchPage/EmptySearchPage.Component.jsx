import React from 'react';

import SearchPage from './SearchPage.Component';

export default class EmptySearchPage extends SearchPage {
    
}