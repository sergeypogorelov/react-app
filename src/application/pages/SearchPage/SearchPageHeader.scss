@import '../../common/scss/base.scss';

.wrapper-header {
    height: 300px;
    background: url('/assets/img/search-background.jpg') -50px -120px;    
}

.wrapper-header-background {
    height: 300px;
    background: rgba(0, 0, 0, 0.8);
}

.wrapper-header-content {
    width: 600px;
    margin: 0 auto;
}

.wrapper-header-content-title {
    margin: 0;
    padding-top: 10px;
    font-size: 15px;
    color: $primaryTitleColor;
}

.wrapper-header-content-title > .link {
    padding: 5px 15px;
    float: right;
    font-size: 15px;
    color: $primaryBtnBgColor;
    background: $defaultBtnBgColor;
    border: none;
    border-radius: 2px;
}

/*
    Search Form
*/

.search-top {
    margin-top: 60px;
}

.search-top-title {
    color: $defaultTitleColor;
}

.search-top-input {
    width: 100%;
    height: 35px;
    padding-left: 10px;
    border: none;
    color: $defaultTxtColor;
    background-color: $defaultBgColor;
    border-bottom: 2px solid $primaryBorderColor;
}

.search-label, .search-button {
    float: left;
}

.search-label {
    margin-top: 30px;
    font-size: 15px;
    font-weight: bold;
    color: $defaultTxtColor;
}

.search-button {
    margin-top: 25px;
    margin-left: 15px;
    padding: 5px 15px;
    font-size: 15px;
    color: $primaryBtnTxtColor;
    background: $primaryBtnBgColor;
    border: none;
    border-radius: 2px;
}

.search-button.active {
    background: $disabledBtnBgColor;
}

.search-submit {
    margin-top: 20px;
    margin-left: 15px;
    padding: 10px 50px;
    float: right;
    font-size: 18px;
    color: $primaryBtnTxtColor;
    background: $primaryBtnBgColor;
    border: none;
    border-radius: 2px;
}