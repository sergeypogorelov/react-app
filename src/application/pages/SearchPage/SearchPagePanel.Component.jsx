import React from 'react';

import './SearchPagePanel.scss';

export default class SearchPagePanel extends React.Component {
    
    render() {
        return (
            <div className="info clearfix">
                <div className="info-content">
                    7 movies found
                </div>
                <div className="info-sort">
                    Sort by
                    <a className="info-sort-link active">release data</a>
                    <a className="info-sort-link">rating</a>
                </div>
            </div>
        );
    }

}