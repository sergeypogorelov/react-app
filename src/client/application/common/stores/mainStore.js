import { applyMiddleware, createStore } from 'redux';
import reduxPromiseMiddleware from 'redux-promise-middleware';

import reducer from '../reducers';

export function configureStore(params) {
    if (!params) {
        params = {};
    }

    let { initialState, middleware } = params;

    if (initialState) {
        if (middleware) {
            console.log('Store is full.');
            return createStore(reducer, initialState, applyMiddleware(middleware, reduxPromiseMiddleware()));
        }

        console.log('Store with initialState.');
        return createStore(reducer, initialState, applyMiddleware(reduxPromiseMiddleware()));
    }

    if (middleware) {
        console.log('Store with middleware.');
        return createStore(reducer, applyMiddleware(middleware, reduxPromiseMiddleware()));
    }

    console.log('Store is empty.');
    return createStore(reducer, applyMiddleware(reduxPromiseMiddleware()));
}

export default configureStore();