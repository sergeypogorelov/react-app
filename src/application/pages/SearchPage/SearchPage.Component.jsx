import React from 'react';
import { Redirect } from 'react-router-dom';

import './SearchPage.scss';

import FilmsStorage from '../../storages/FilmsStorage';

import FilmsList from '../../common/components/FilmsList/FilmsList.Component';
import Footer from '../../common/components/Footer/Footer.Component';

import SearchPageHeader from './SearchPageHeader.Component';
import SearchPagePanel from './SearchPagePanel.Component';

export default class SearchPage extends React.Component {

    constructor() {
        super();
        this.state = {
            films: [],
            filmsLoaded: false
        };
    }

    componentWillMount() {
        this.state.searchQuery = this.props.match.params.query || '';
    }

    componentDidMount() {
        this.loadFilms(this.state.searchQuery);
    }

    searchFormHandler(searchParams) {
        if (this.state.searchQuery !== searchParams.searchQuery) {
            this.props.history.push('/search/' + searchParams.searchQuery);  
        }

        this.loadFilms(searchParams.searchQuery);
    }

    searchPanelHandler() {
        this.loadFilms(this.state.searchQuery);
    }

    render() {
        return (
            <div className="wrapper">
                <SearchPageHeader query={this.state.searchQuery} onSubmit={this.searchFormHandler.bind(this)} />
                <div className="wrapper-content">
                    <SearchPagePanel films={this.state.films} filmsLoaded={this.state.filmsLoaded} onSortChange={this.searchPanelHandler.bind(this)} />
                    <FilmsList films={this.state.films} filmsLoaded={this.state.filmsLoaded} />
                </div>
                <Footer />
            </div>
        );
    }

    loadFilms(query) {
        this.changeState({ searchQuery: query, filmsLoaded: false });
        FilmsStorage.searchItem(query).then((films) => {
            this.changeState({ films, filmsLoaded: true });
        });
    }

    changeState(state) {
        let obj = {};
        Object.assign(obj, this.state, state);
        this.setState(obj);
    }

}