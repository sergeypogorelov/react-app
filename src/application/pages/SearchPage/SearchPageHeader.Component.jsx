import React from 'react';

import './SearchPageHeader.scss';

import FilmsStorage from '../../storages/FilmsStorage';

export default class SearchPageHeader extends React.Component {

    constructor() {
        super();
        this.state = {
            searchQuery: '',
            searchType: FilmsStorage.searchType
        };
    }

    componentWillMount() {
        this.state.searchQuery = this.props.query;
    }

    setSearchByTitle(ev) {
        ev.preventDefault();

        FilmsStorage.searchType = 'title';
        this.setState({
            searchQuery: this.state.searchQuery,
            searchType: 'title'
        });
    }

    setSearchByDirector(ev) {
        ev.preventDefault();

        FilmsStorage.searchType = 'director';
        this.setState({
            searchQuery: this.state.searchQuery,
            searchType: 'director'
        });
    }

    submitHandler(ev) {
        ev.preventDefault();
        this.props.onSubmit({ searchQuery: this.state.searchQuery });
    }

    searchQueryHandler(ev) {
        this.changeState({
            searchQuery: ev.target.value
        });
    }

    render() {
        return (
            <div className="wrapper-header">
                <div className="wrapper-header-background">
                   <div className="wrapper-header-content">
                        <h1 className="wrapper-header-content-title">Netflix Roulete</h1>
                        <div className="wrapper-header-content-search">
                            <form className="search" onSubmit={this.submitHandler.bind(this)}>
                                <div className="search-top">
                                    <h3 className="search-top-title">FIND YOUR MOVIE</h3>
                                    <input className="search-top-input" value={this.state.searchQuery} onChange={this.searchQueryHandler.bind(this)} />
                                </div>
                                <div className="search-bottom clearfix">
                                    <p className="search-label">SEARCH BY</p>
                                    <input className={ this.state.searchType === 'title' ? 'search-button active' : 'search-button' } onClick={this.setSearchByTitle.bind(this)} type="button" value="TITLE" />
                                    <input className={ this.state.searchType === 'director' ? 'search-button active' : 'search-button' } onClick={this.setSearchByDirector.bind(this)} type="button" value="DIRECTOR" />
                                    <input className="search-submit" type="submit" value="SEARCH" />
                                </div>
                            </form>
                        </div>
                   </div>
                </div>
            </div>
        );
    }

    changeState(state) {
        let obj = {};
        Object.assign(obj, this.state, state);
        this.setState(obj);
    }

}