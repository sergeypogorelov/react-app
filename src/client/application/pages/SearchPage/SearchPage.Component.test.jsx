import React from 'react';

import renderer from 'react-test-renderer';

import mainStore from '../../common/stores/mainStore';
import { SearchPage } from './SearchPage.Component';

jest.mock('filmsActions');

test('Component SearchPage works fine.', () => {

    const props = {
        films: [],

        filmsLoaded: false,
        filmsLoading: false,
        filmsNotLoaded: false,

        match: {
            params: {
                query: ''
            }
        },
        location: {
            search: ''
        },
        dispatch: mainStore.dispatch
    };

    let component = renderer.create(
        <SearchPage { ...props } />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    props.filmsNotLoaded = true;

    component = renderer.create(
        <SearchPage { ...props } />
    );

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});