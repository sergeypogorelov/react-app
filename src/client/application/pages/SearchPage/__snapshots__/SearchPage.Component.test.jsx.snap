// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchPage works fine. 1`] = `
<div
  className="wrapper"
>
  <div
    className="wrapper-header"
  >
    <div
      className="wrapper-header-background"
    >
      <div
        className="wrapper-header-content"
      >
        <h1
          className="wrapper-header-content-title"
        >
          Netflix Roulete
        </h1>
        <div
          className="wrapper-header-content-search"
        >
          <form
            className="search"
            onSubmit={[Function]}
          >
            <div
              className="search-top"
            >
              <h3
                className="search-top-title"
              >
                FIND YOUR MOVIE
              </h3>
              <input
                className="search-top-input"
                onChange={[Function]}
                value=""
              />
            </div>
            <div
              className="search-bottom clearfix"
            >
              <p
                className="search-label"
              >
                SEARCH BY
              </p>
              <input
                className="search-button active"
                onClick={[Function]}
                type="button"
                value="TITLE"
              />
              <input
                className="search-button"
                onClick={[Function]}
                type="button"
                value="DIRECTOR"
              />
              <input
                className="search-submit"
                type="submit"
                value="SEARCH"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="wrapper-content"
  >
    <div
      className="info"
    />
    <h1>
      No films found
    </h1>
  </div>
  <div
    className="wrapper-footer"
  >
    <p
      className="wrapper-footer-content"
    >
      Netflix Roulete
    </p>
  </div>
</div>
`;

exports[`Component SearchPage works fine. 2`] = `
<div
  className="wrapper"
>
  <div
    className="wrapper-header"
  >
    <div
      className="wrapper-header-background"
    >
      <div
        className="wrapper-header-content"
      >
        <h1
          className="wrapper-header-content-title"
        >
          Netflix Roulete
        </h1>
        <div
          className="wrapper-header-content-search"
        >
          <form
            className="search"
            onSubmit={[Function]}
          >
            <div
              className="search-top"
            >
              <h3
                className="search-top-title"
              >
                FIND YOUR MOVIE
              </h3>
              <input
                className="search-top-input"
                onChange={[Function]}
                value=""
              />
            </div>
            <div
              className="search-bottom clearfix"
            >
              <p
                className="search-label"
              >
                SEARCH BY
              </p>
              <input
                className="search-button active"
                onClick={[Function]}
                type="button"
                value="TITLE"
              />
              <input
                className="search-button"
                onClick={[Function]}
                type="button"
                value="DIRECTOR"
              />
              <input
                className="search-submit"
                type="submit"
                value="SEARCH"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="wrapper-content"
  >
    <div
      className="info"
    />
    <h1>
      Something's wrong
    </h1>
  </div>
  <div
    className="wrapper-footer"
  >
    <p
      className="wrapper-footer-content"
    >
      Netflix Roulete
    </p>
  </div>
</div>
`;
